### DISCUSSION TITLE
Unable to override variables in entry.js file

### DISCUSSION THREAD

**Discussion by ghost** [nuxt/nuxt#23469] [General]
Hi, I am not able to override env variables that are defined in `entry.js` file. 
To make nuxt app flexible I did create a file `envConfig.json` with placeholder values. 

Sample `envConfig.json` file:
```
  "APP_ID": "ph_APP_ID",
  "CLIENT_ID": "ph_CLIENT_ID",
```
Then I build app and dockerize. After build of docker, I execute entrypoint script to replace those placeholders, with values read from env variables. 

```
export ENV=$(printenv | grep ENV | sed 's/^/\ph_/g' | paste -sd,);
ELEMENTS=$(awk -F',' '{ for( i=1; i<=NF; i++ ) print $i }' <<<"$ENV")

for file in /app/.output/public/_nuxt/entry*.js ; do
  for element in $ELEMENTS ; do
    key=$(cut -d "=" -f1 <<< "$element")
    value=$(cut -d "=" -f2 <<< "$element")
    sed -i -e "s|$key|$value|g" "$file"
  done
done
```

Unfortunately, I get error from web console:
` "Uncaught SyntaxError: Unexpected end of input (at entry.9f2a3fdd.js:1280:2575)"`

I think this might be related to **Content-Lenght** response header, as this value is equal to entry.js file before placeholders replacement.  
Can you help with that?

**Comment by BuuBux** [comment]
I guess it is describe here also in terms of mismatch content length

[Complex Runtime Config](https://www.youtube.com/watch?v=vdUZqZk6iXQ&t=370s)

I also struggle with similar issue. Is there any possible way to replace env variables after build from k8s secrets?
